decorator_allowed_groups/models/ir_ui_view.py

```diff
@@ -0,0 +1,20 @@
+# Copyright (C) 2021-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class IrUiView(models.Model):
+    _inherit = "ir.ui.view"
+
+    @api.model
+    def postprocess(self, model, node, view_id, in_tree_view, model_fields):
+        if node.tag == "button" and node.get("name", False):
+            func = getattr(self.env[model], node.get("name"), False)
+            if func:
+                group_xml_ids = getattr(func, "_allowed_groups", False)
+                if group_xml_ids:
+                    node.set("groups", ",".join(group_xml_ids))
```

Comment:
Maybe emit a warning if groups is already set and not equivalent?

Reply:
I thought about that point, and I'm not sure.

Consider you have an Odoo / OCA module with a ``def my_action(self)`` defined, and a button defined in a view with ``groups="module_1._group_A"``.

In a custom module, you want to change the group, so you simply write 

```
    @api.allowed_groups("module_2.group_B")
    def my_action(self):
        return super().my_action(self)
```

If you do so, it will raise a false warning, forcing developpers to overload the view in the custom module, to remove the group. 

I so just added in the ``USAGE.rst`` file this text : 

``
The groups defined in the decorators take precedence over the groups that would be defined in the existing views.
``

What do you think ?

Reply:
My preference would be to emit the warning. It seems to me that the value of this module is mainly for methods introduced in modules that use this decorator in the first place, rather than overriding existing methods, and in that case, aligning the views is the proper thing to do to avoid confusion. But it's just my preference.

Reply:
What about combining the groups in the button and in the decorator, so you can only limit the authorities by either method, but never grant authorities?

Reply:
I would say that is not the way I would expect and want such a mechanism to work. I guess it depends on whether you take a security perspective or a usability perspective.

Reply:
It is what I would expect. But I think I expressed myself poorly. Combining should be interpreted as: when groups are defined on the button and in the decorator, only the groups that are valid in both should be shown. When this leaves no valid groups, at least a warning should be shown, or maybe even an exception thrown.

Reply:
You're right. :+1: 
I added this point to the roadmap. Better to add a warning. We can remove it in a second time, if we consider later, that it is not relevant in some cases.

Reply:
> only the groups that are valid in both should be shown

Ah indeed, I misunderstood. This would improve the usability as well as the security. :+1: